src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -327,7 +328,28 @@ async def generate_lockfiles_goal(
     union_membership: UnionMembership,
     generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
     python_setup: PythonSetup,
+    python_repos: PythonRepos,
 ) -> GenerateLockfilesGoal:
+    failure_instructions = (
+        "If lockfile generation fails, you can either disable lockfiles by setting "
+        "`[tool].lockfile = '<none>'`, e.g. setting `[black].lockfile`, or you can manually "
+        "generate lockfiles, e.g. by using pip-compile or `pip freeze`. If manually generating "
+        "lockfiles, you should set `[python-setup].invalid_lockfile_behavior = 'ignore'."
```

Comment:
@chrisjrn and I discussed that it could be possible to manually maintain a lockfile and set the Pants lockfile metadata yourself. But he intuited that is confusing / error-prone, and he suggested we simply don't use that mechanism if you're manually generating lockfiles. I now agree with him.